### PR TITLE
UI changes and Course Critique Integration

### PR DESCRIPTION
**Pull request by abhitirumala**
Hi Jason,

Thank you for letting my use your code!

Besides the website, if you accept the PR, I will also move the domain gtscheduler.tech to this repo and add you to the good analytics portal.

Hope you like what I added!

### CODE REVIEW

**Review by 64json — CHANGES_REQUESTED** [review]
Hey @abhitirumala, thank you for the PR.

However, I won't be able to review this PR since several thousand lines of [**diff**](https://github.com/64json/gt-scheduler/pull/3/files) are due to different code styles as below:
Screen Shot 2020-06-15 at 1 48 21 AM

The ideal way would be to conform with the original code style from the beginning. Since quite a lot of logical changes have been made on top of the reformatted code, I suggest you to do the following instead:
1. Create a branch from the current `master` branch of this repo.
2. Run code formatting (using the same setup as you used for this PR).
3. Create a PR only containing code style changes.

Once I merge that PR into `master` and recreate diff on this PR, I'll be able to review only the logical changes.

File: src/beans/canvasjs-2.3.2/license.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+*
+* @preserve CanvasJS HTML5 & JavaScript Charts - v2.3.2 GA - https://canvasjs.com/ 
+* Copyright 2018 fenopix
+*
+*  --------------------- License Information --------------------
+* CanvasJS is a commercial product which requires purchase of license. Without a commercial license you can use it for evaluation purposes for upto 30 days. Please refer to the following link for further details.
+*     https://canvasjs.com/license/
```
Comment (64json) [inline src/beans/canvasjs-2.3.2/license.txt]: I don't think I have money to afford this library unless we are going to remove this feature after 30 days.

File: CNAME [hunk 2]
```diff
@@ -0,0 +1 @@
+gtscheduler.tech
```
Comment (64json) [inline CNAME]: `CNAME` should be removed.

File: src/beans/fetchCourseCritique.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import cheerio from 'cheerio';
+import $ from 'jquery';
```
Comment (64json) [inline src/beans/fetchCourseCritique.js]: jQuery is usually discouraged to be used with a front-end framework like React or Angular. If you only used it for fetching data, I recommend you to use [axios](https://github.com/axios/axios) instead.

https://github.com/64json/gt-scheduler/blob/1d5d8a715b55b1b14141126cab151fc04087f911/src/components/App/index.jsx#L34

File: src/components/ActionRow/stylesheet.scss [hunk 4]
```diff
@@ -26,7 +27,66 @@
...
+      }
+    }
+  }
+
+  &:hover .actions {
+    visibility: visible;
+  }
+}
+
+.App.dark .ActionRow {
```
Comment (64json) [inline src/components/ActionRow/stylesheet.scss]: Most of the properties of `.App.dark .ActionRow` are the duplicates of `.App.light .ActionRow`.

`.ActionRow` should have common properties that don't depend on themes, and `.App.light .ActionRow` and `.App.dark .ActionRow` should only contain theme-specific properties.

Comment (64json) [reply src/components/ActionRow/stylesheet.scss]: It goes the same for any other stylesheets.

**Review by 64json — CHANGES_REQUESTED** [review]

File: src/beans/canvasjs-2.3.2/license.txt (hunk 1 above)
Comment (64json) [reply src/beans/canvasjs-2.3.2/license.txt]: [vx](https://github.com/hshoff/vx) would be a good alternative.

File: public/index.html [hunk 5]
```diff
@@ -1,28 +1,34 @@
...
       window.dataLayer = window.dataLayer || [];
-
       function gtag() {
         dataLayer.push(arguments);
       }
-
       gtag('js', new Date());
 
-      gtag('config', 'UA-128271224-1');
+      gtag('config', 'UA-156655464-1');
```
Comment (64json) [inline public/index.html]: I would still use the original tracking code.

File: public/index.html [hunk 6]
```diff
@@ -32,23 +38,30 @@
...
+      property="og:description"
+      content="Georgia Tech Scheduler V2 lets you find the schedule that fits you best among all the possible combinations of courses."
+    />
+  </head>
+  <body>
+    <noscript>
+      You need to enable JavaScript to run this app.
+    </noscript>
+    <div id="root"></div>
+    <!--
```
Comment (64json) [inline public/index.html]: Revert `title`, `og:image`, `og:title`, and `og:url`. For `author`, you can change to `Jinseo Park and [Your Name]`.

File: public/manifest.json [hunk 7]
```diff
@@ -1,9 +1,9 @@
 {
-  "short_name": "GT Scheduler",
-  "name": "GT Scheduler",
+  "short_name": "GT Scheduler V2",
+  "name": "GT Scheduler V2",
```
Comment (64json) [inline public/manifest.json]: Revert this as well please.

File: README.md [hunk 8]
```diff
@@ -1,4 +1,41 @@
...
+- Add "No Class" Time Blocks
+- Comprehensive Course Difficulty
+- More Graphs and Data Visualizations
+- More UI improvements!
+
+![GT Scheduler Split Theme](https://raw.githubusercontent.com/abhitirumala/gt-scheduler-v2/master/public/image.jpg)
+
+## Credits
+
+Credits to **Jason Park** (https://github.com/64json/gt-scheduler) for initially developing the popular web application and helping the Georgia Tech Community in making course registration an easier task.
```
Comment (64json) [inline README.md]: Revert the entire `README.md`. If you wish to include change log, do so in `CHANGELOG.md`. For the coming-soon-features, we can create a GitHub issue for each of them to have a discussion thread.

File: package.json [hunk 9]
```diff
@@ -1,29 +1,43 @@
 {
-  "name": "gt-scheduler",
+  "name": "gt-scheduler-v2",
   "version": "0.1.0",
   "private": true,
-  "homepage": "https://jasonpark.me/gt-scheduler",
+  "homepage": "https://gtscheduler.tech/",
```
Comment (64json) [inline package.json]: Revert `name` and `homepage`, and increase version to `0.2.0`. Please refer to [Semantic Versioning](https://semver.org) how a version number is determined.

**Review by 64json — CHANGES_REQUESTED** [review]

File: package.json [hunk 10]
```diff
@@ -1,29 +1,43 @@
...
+    "react-resize-panel": "^0.3.5",
+    "react-scripts": "^3.4.1",
+    "react-virtualized": "^9.21.2",
+    "redux": "^4.0.5",
+    "redux-actions": "^2.6.5",
+    "request": "^2.88.2",
+    "rmp-api": "0.0.1-security",
+    "semantic-ui-react": "^0.88.2",
+    "typescript": "^3.9.3",
+    "underscore": "^1.10.2"
```
Comment (64json) [inline package.json]: I see a lot of packages that are potentially not in use:
- `@emotion/core`
- `@emotion/styled`
- `@progress/kendo-react-popup`
- `ajv`
- `cors-anywhere`
- `request`
- `rmp-api`
- `semantic-ui-react`
- `typescript`
- `underscore`

They should be removed.

**Review by 64json — CHANGES_REQUESTED** [review]

File: CNAME (hunk 2 above)
Comment (abhitirumala) [reply CNAME]: Removed.

**Pull request closed without merging**